.github/workflows/lint.yaml

```diff
@@ -85,13 +85,18 @@ jobs:
         key: macos-12-build-cache-${{ hashFiles('./scripts/builder_setup.sh') }}
 
     - name: Clone datadog-agent repository
+      uses: nick-fields/retry@v3
```

Comment:
```suggestion
      uses: nick-fields/retry@7152eba30c6575329ac0576536151aca5a72780e # v3.0.0
```
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏It'd be a bit safer imo to pin the action to its commit hash as we're doing with the rest of the GHA. WDYT?